src/main/scala/mesosphere/marathon/core/storage/simple/SimplePersistenceStore.scala

```diff
@@ -0,0 +1,159 @@
+package mesosphere.marathon
+package core.storage.simple
+
+import akka.stream.scaladsl.Flow
+import akka.util.ByteString
+import akka.{Done, NotUsed}
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.marathon.core.storage.simple.PersistenceStore._
+import mesosphere.marathon.metrics.{Counter, Metrics, ServiceMetric}
+import org.apache.zookeeper.KeeperException.NoNodeException
+
+import scala.collection.JavaConverters
+import scala.compat.java8.FutureConverters._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Try}
+
+/**
+  * An implementation of the [[PersistenceStore]] interface.
+  *
+  * It utilises Apache Curator async methods under the hood from the org.apache.curator.curator-x-async library. This
+  * is similar to synchronous methods with [[org.apache.curator.framework.api.Backgroundable]] callback however, it
+  * uses a somewhat cleaner [asynchronous DSL](https://curator.apache.org/curator-x-async/index.html).
+  *
+  * Create, update, read and delete operation are offered as akka-stream Flows. Every Flow will offer an output signalising
+  * that the input element was handled successfully. E.g. a [[create]] flow takes a stream of nodes to store and outputs
+  * a stream of node paths indicating that the node was successfully stored.
+  *
+  * For CRUD operations the submitted stream elements are handled with configured parallelism. For create and update
+  * operations the implementation guaranties that if multiple operations update the same path, the order of writes is the
+  * same as in the input stream.
+  *
+  * Transaction support is implemented and four transaction operations are supported: [[CreateOp]], [[UpdateOp]],
+  * [[DeleteOp]] and [[CheckOp]]. All transaction operations has to be successful for the transaction to commit successfully.
+  *
+  * @param factory an instance of [[AsyncCuratorBuilderFactory]]
+  * @param parallelism parallelism level for CRUD operations
+  * @param ec execution context
+  */
+class SimplePersistenceStore(factory: AsyncCuratorBuilderFactory, parallelism: Int = 10)(implicit ec: ExecutionContext)
+  extends PersistenceStore with StrictLogging {
+
+  // format: OFF
+  private[this] val createMetric          = Metrics.counter(ServiceMetric, getClass, "create")
```

Comment:
What about metrics for op times and rates?

Reply:
Rate could be derived from the counter but op time is a good thing to have. If you know how to measure stream's stage execution time elegantly - I'm all ears. Otherwise, I'd rather implement it in the next PR.